osf/models/user_message.py

```diff
@@ -0,0 +1,111 @@
+from typing import Type
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from .base import BaseModel, ObjectIDMixin
+from website.mails import send_mail, USER_MESSAGE_INSTITUTIONAL_ACCESS_REQUEST
+
+
+class MessageTypes(models.TextChoices):
+    """
+    Enumeration of the different user-to-user message types supported by UserMessage.
+
+    Notes:
+        Message types should be limited to direct communication between two users.
+        These may include cases where the sender represents an organization or group,
+        but they must not involve bulk messaging or group-wide notifications.
+    """
+    # Admin-to-member communication within an institution.
+    INSTITUTIONAL_REQUEST = ('institutional_request', 'INSTITUTIONAL_REQUEST')
+
+    @classmethod
+    def get_template(cls: Type['MessageTypes'], message_type: str) -> str:
+        """
+        Retrieve the email template associated with a specific message type.
+
+        Args:
+            message_type (str): The type of the message.
+
+        Returns:
+            str: The email template string for the specified message type.
+        """
+        return {
+            cls.INSTITUTIONAL_REQUEST: USER_MESSAGE_INSTITUTIONAL_ACCESS_REQUEST
+        }[message_type]
+
+
+class UserMessage(BaseModel, ObjectIDMixin):
+    """
+    Represents a user-to-user message, potentially sent on behalf of an organization or group.
+
+    Attributes:
+        sender (OSFUser): The user who initiated the message.
+        recipient (OSFUser): The intended recipient of the message.
+        message_text (str): The content of the message being sent.
+        message_type (str): The type of message, e.g., 'institutional_request'.
+        institution (Institution): The institution linked to the message, if applicable.
+    """
+    sender = models.ForeignKey(
+        'OSFUser',
+        on_delete=models.CASCADE,
+        related_name='sent_user_messages',
+        help_text='The user who sent this message.'
+    )
+    recipient = models.ForeignKey(
+        'OSFUser',
+        on_delete=models.CASCADE,
+        related_name='received_user_messages',
+        help_text='The user who received this message.'
```

Comment:
```suggestion
        help_text='The recipient of this message.'
```

Just because technically they haven't received the message at the time of instantiation in the database. This is, of course, extremely minor.